### PR TITLE
Update Paddle OP support via hackathon in PRC

### PR DESCRIPTION
**Pull request by OpenVINO-dev-contest**
### Details:

-     Paddle "not_equal" op upgrade
-     Paddle "expand" op upgrade
-     Paddle "topk" op upgrade
-     Paddle "fill_zeros_like" op upgrade
-     Paddle "gather_tree" op upgrade

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Comment by ceciliapeng2011** [comment]
@luo-cheng2021 @liubo-intel @ilyachur @mvafin please help review. They are contributed from openvino community.

**Review by ceciliapeng2011 — CHANGES_REQUESTED** [review]

File: ngraph/test/frontend/paddlepaddle/test_models/gen_scripts/generate_top_k.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+data_type = 'float32'
+
+
+def top_k(name: str, x, k: int):
+
+    pdpd.enable_static()
+
+    with pdpd.static.program_guard(pdpd.static.Program(), pdpd.static.Program()):
+        node_x = pdpd.static.data(name='x', shape=x.shape, dtype='float32')
+        value, indices = fluid.layers.topk(node_x, k=k, name="top_k")
```
Comment (ceciliapeng2011) [inline ngraph/test/frontend/paddlepaddle/test_models/gen_scripts/generate_top_k.py:19]: thanks for add test case for legacy topk. Can we merge it with topk_v2 into a single test file?

File: ngraph/frontend/paddlepaddle/src/op/top_k.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+//
+
+#include <ngraph/opsets/opset6.hpp>
+#include <node_context.hpp>
+
+namespace ov {
+namespace frontend {
+namespace pdpd {
+namespace op {
+NamedOutputs top_k(const NodeContext& node) {
```
Comment (ceciliapeng2011) [inline ngraph/frontend/paddlepaddle/src/op/top_k.cpp:12]: Can we merge it with topk_v2? topk is a special case of topk_v2. We'd better take care of them together to reuse code.

File: ngraph/frontend/paddlepaddle/src/op/top_k_v2.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    int64_t axis = node.get_attribute<int64_t>("axis", -1);
+    bool sorted = node.get_attribute<bool>("sorted", true);
+    bool largest = node.get_attribute<bool>("largest", true);
+    const element::Type& index_element_type = element::i64;
+
+    auto sort_type = sorted ? ngraph::opset6::TopK::SortType::SORT_VALUES : ngraph::opset6::TopK::SortType::NONE;
+    auto mode = largest ? ngraph::opset6::TopK::Mode::MAX : ngraph::opset6::TopK::Mode::MIN;
+
+    auto node_topk =
+        std::make_shared<ngraph::opset6::TopK>(x, k_expected_node, axis, mode, sort_type, index_element_type);
```
Comment (ceciliapeng2011) [inline ngraph/frontend/paddlepaddle/src/op/top_k_v2.cpp:32]: again, please use defaultopset instead of specify opset number here.

**Review by ceciliapeng2011 — CHANGES_REQUESTED** [review]

File: src/frontends/paddlepaddle/src/op/expand.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+#include <node_context.hpp>
+#include <paddlepaddle_frontend/utility.hpp>
+
+#include "default_opset.hpp"
+using namespace default_opset;
+namespace ov {
+namespace frontend {
+namespace pdpd {
+namespace op {
+NamedOutputs expand(const NodeContext& node) {
```
Comment (ceciliapeng2011) [inline src/frontends/paddlepaddle/src/op/expand.cpp:14]: We can use a function to handle the common part for expand and expand_v2.

**Review by ceciliapeng2011 — CHANGES_REQUESTED** [review]

**Review by ceciliapeng2011 — COMMENTED** [review]
Why there are 163 files changed? Could you please rebase and exclude files that are not irrelated? 
Once the conversation is resolved or outdated, could you please mark it "resolved"?
@Ethan

**Comment by Openvino-dev-contest** [comment]
> Why there are 163 files changed? Could you please rebase and exclude files that are not irrelated? Once the conversation is resolved or outdated, could you please mark it "resolved"? @Ethan

sorry, problem has fixed, pls review again.

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Pull request closed without merging**